# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/NodeChannel.cpp
# Commit: 451508713a62
# Full Hash: 451508713a625beaaa9fcbbf06c65ca312dc34fb
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 878722e87e62 -r 451508713a62 ipc/glue/NodeChannel.cpp
--- a/ipc/glue/NodeChannel.cpp	Tue Jun 22 18:17:23 2021 +0000
+++ b/ipc/glue/NodeChannel.cpp	Tue Jun 22 18:17:23 2021 +0000
@@ -117,7 +117,7 @@
 
   if (!mClosed) {
     mChannel->Close();
-    MOZ_ALWAYS_TRUE(this == mChannel->set_listener(mExistingListener));
+    mChannel->set_listener(mExistingListener);
   }
   mClosed = true;
 }